arch/risc-v/src/rv64gc/crt0.c

```diff
@@ -0,0 +1,136 @@
+ ****************************************************************************/
+
+int main(int argc, char *argv[]);
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_signal_handler
+ *
+ * Description:
+ *   This function is the user-space, signal handler trampoline function.  It
+ *   is called from up_signal_dispatch() in user-mode.
+ *
+ * Input Parameters:
+ *   a0 = sighand
+ *     The address user-space signal handling function
+ *   a1-a3 = signo, info, and ucontext
+ *     Standard arguments to be passed to the signal handling function.
+ *
+ * Returned Value:
+ *   None.  This function does not return in the normal sense.  It returns
+ *   via the SYS_signal_handler_return (see svcall.h)
+ *
+ ****************************************************************************/
+
+static void sig_trampoline(void) naked_function;
+static void sig_trampoline(void)
+{
+  __asm__ __volatile__
+  (
+    " addi sp, sp, -16\n"   /* Save ra on the stack */
+    " sd   ra, 8(sp)\n"
+    " mv   t0, a0\n"        /* t0=sighand */
+    " mv   a0, a1\n"        /* a0=signo */
+    " mv   a1, a2\n"        /* a1=info */
+    " mv   a2, a3\n"        /* a2=ucontext */
+    " jalr t0\n"            /* Call the signal handler (modifies ra) */
+    " ld   ra, 8(sp)\n"     /* Recover ra in sp */
+    " addi sp, sp, 16\n"
+    " li   a0, %0\n"        /* SYS_signal_handler_return */
+    " ecall\n"              /* Return from the SYSCALL */
+    " nop\n"
+    :
+    : "i" (SYS_signal_handler_return)
+    :
+  );
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: _start
+ *
+ * Description:
+ *   This function is the low level entry point into the main thread of
+ *   execution of a task.  It receives initial control when the task is
+ *   started and calls main entry point of the newly started task.
+ *
+ * Input Parameters:
+ *   argc - The number of parameters being passed.
+ *   argv - The parameters being passed. These lie in kernel-space memory
+ *     and will have to be reallocated  in user-space memory.
+ *
+ * Returned Value:
+ *   This function should not return.  It should call the user-mode start-up
+ *   main() function.  If that function returns, this function will call
+ *   exit.
+ *
+ ****************************************************************************/
+
+void _start(int argc, FAR char *argv[])
+{
+  int ret;
+
+  /* Initialize the reserved area at the beginning of the .bss/.data region
+   * that is visible to the RTOS.
+   */
+  ARCH_DATA_RESERVE->ar_sigtramp = (addrenv_sigtramp_t)sig_trampoline;
```

Comment:
This predefined vaddr is used to share the signal trampoline (as well as the heap structure) between the user and kernel. This does not work for us, and something else needs to be figured out.